docs/concurrency.org

```diff
@@ -0,0 +1,79 @@
+Note: il se peut qu je dise des trucs faux car la plupart des ces termes sont complétement nouveaux, donc si tu vois des trucs louches, cest assez probable que tu aies un bon feeling.
+
+Je dois voir Alberto lors de ma prochaine séance pour lui poser plusieurs questsions sur ces différents termes etc
+
+** Definitions
+Coroutine (cooperative multitasking): 
+- a coroutine itself is not concurrent !, it allows to manage concurrent tasks in the same way as callbacks for example
+- single threaded
+- allows to write code in a sequential manner
+- allows not to have context switch
+- 2 types of coroutine: symetric vs asymetric (relative to the control-transfer mechanism)
```

Comment:
Maybe explain the difference between symmetric and asymmetric

Reply:
done, tell me what you think (go around line 16 to see)